File: src/main/java/com/milaboratory/mixcr/trees/TreeWithMeta.java
package com.milaboratory.mixcr.trees;

public class TreeWithMeta {
    private final Tree<CloneOrFoundAncestor> tree;
    private final RootInfo rootInfo;
    private final TreeWithMetaBuilder.TreeId treeId;

    public TreeWithMeta(
            Tree<CloneOrFoundAncestor> tree,
            RootInfo rootInfo,
            TreeWithMetaBuilder.TreeId treeId
    ) {
        this.tree = tree;
        this.rootInfo = rootInfo;
        this.treeId = treeId;
    }

    public TreeWithMetaBuilder.TreeId getTreeId() {
        return treeId;
    }

    public Tree<CloneOrFoundAncestor> getTree() {
        return tree;
    }

    public RootInfo getRootInfo() {
        return rootInfo;
    }
}

File: src/main/kotlin/com/milaboratory/mixcr/trees/TreeWithMeta.kt
package com.milaboratory.mixcr.trees

import com.milaboratory.mixcr.trees.TreeWithMetaBuilder.TreeId

class TreeWithMeta(
    val tree: Tree<CloneOrFoundAncestorOld>,
    val rootInfo: RootInfo,
    val treeId: TreeId
)
